src/workers/parser/mapOriginalExpression.js

```diff
@@ -81,22 +77,67 @@ export default function mapOriginalExpression(
     }
   }
 
-  if (replacements.size === 0) {
-    // Avoid the extra code generation work and also avoid potentially
-    // reformatting the user's code unnecessarily.
-    return expression;
+  return replacements;
+}
+
+export default function mapOriginalExpression(
+  expression: string,
+  mappings: {
+    [string]: string | null
   }
+): string {
+  const ast = parseScript(expression);
+  const replacements = getReplacements(ast, mappings)
 
+  let didUpdate = false;
   t.traverse(ast, (node, ancestors) => {
-    if (!t.isIdentifier(node) && !t.isThisExpression(node)) {
+    const parent = ancestors[ancestors.length - 1];
+    if (!parent) {
       return;
     }
+    const parentNode = parent.node;
+
+    if (replacements.size > 0 && (t.isIdentifier(node) || t.isThisExpression(node))) {
+      const replacement = replacements.get(locationKey(node.loc.start));
+      if (replacement) {
+        didUpdate = true;
+        replaceNode(ancestors, t.cloneNode(replacement));
+      }
+    }
+
 
-    const replacement = replacements.get(locationKey(node.loc.start));
-    if (replacement) {
-      replaceNode(ancestors, t.cloneNode(replacement));
+    if (t.isVariableDeclaration(node) && !t.isBlockStatement(parentNode)) {
```

Comment:
💯 Though I think it could use some comments to clarify for those unfamiliar with the `t.*` syntax